src/rgw/rgw_admin.cc

```diff
@@ -288,6 +288,7 @@ void usage()
   cout << "   --max-buckets             max number of buckets for a user\n";
   cout << "   --admin                   set the admin flag on the user\n";
   cout << "   --system                  set the system flag on the user\n";
+  cout << "   --cache                   set the cache flag on the user\n";
```

Comment:
i agree that we should require a 'special' kind of user to accept the `X-Amz-Cache` header because of the security implications. but i see this more of a 'capability', which could be covered by user caps instead of adding a new generic flag:
```
radosgw-admin caps add --uid cacheuser --caps amz-cache=read
```
and checked with:
```c++
const auto& caps = s->user->get_caps();
int ret = caps.check_cap("amz-cache", RGW_CAP_READ); // returns 0 or -EPERM
```

Reply:
Good idea I will check it, thank you

Reply:
that should work once you add "amz-cache" to the list in `RGWUserCaps::is_valid_cap_type()`

Reply:
Resolved